### DISCUSSION TITLE
Spacenavd + wine how to use?

### DISCUSSION THREAD

**Discussion by ARDev1161** [FreeSpacenav/spacenavd#109] [Questions and support]
I use spnav on my host pc. But I work in CAD system that installed in Wine(VirtualBox give really big delay from Space Mouse). How can I force data from host to Wine?

**Comment by jtsiomb** [comment]
Assuming applications on windows link a 3Dconnexion dll to get 6dof input, you'd need to write some kind of "proxy" dll that implements the 3Dconnexion windows API, while getting input from spacenavd. Then you'd drop this into the directory of the application, probably replacing the original dll, and it should work.

If the 3Dconnexion API is implemented as a static library it's more complicated. You'd need to reverse-engineer the protocol used to communicate between the 3Dconnexion service (or whatever it is, assuming it's a windows service), and the client code in the 3Dconnexion static client library, and write a proxy service that again talks to spacenavd, and provides the same protocol.

If you decide to tackle any of that let me know how it goes.
